### PR TITLE
feat return errors in the format that pro agent expects

### PR DESCRIPTION
**Pull request by necipallef**
Screen Shot 2022-07-04 at 00 19 21

### CODE REVIEW

**Comment by necipallef** [comment]
Fixes #1

File: src/utils/createErrorResponse.ts [hunk 1]
```diff
@@ -1,13 +1,59 @@
-export function createErrorResponse(error: string | Error | unknown) {
-  let reason
+export interface Response {
+  v: '2'
```
Comment (sshelomentsev) [inline src/utils/createErrorResponse.ts:2]: What is the v2?

File: src/utils/createErrorResponse.ts [hunk 2]
```diff
@@ -1,13 +1,59 @@
...
+function generateRandomString(length: number): string {
+  let result = ''
+  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'
+  for (let i = 0; i < length; i++) {
+    result += characters.charAt(Math.floor(Math.random() * characters.length))
+  }
+  return result
+}
+
+function generateRequestUniqueId(): string {
```
Comment (sshelomentsev) [inline src/utils/createErrorResponse.ts:36]: How do we plan to use this request id?

File: src/utils/createErrorResponse.ts (hunk 1 above)
Comment (necipallef) [reply src/utils/createErrorResponse.ts:2]: `v` field is used to separate API v1 and API v2 responses by the Pro Agent.

File: src/utils/createErrorResponse.ts (hunk 2 above)
Comment (necipallef) [reply src/utils/createErrorResponse.ts:36]: This `requestId` has two purposes:
1- Separate the requests that reach and doesn't reach CoreAPI. With `cfi-` infix, a customer is able to tell that the response is generated by the CF Worker
2- Later on, if we can store logs, we can reach a specific request's logs by using `requestId` as an identifier.

**Pull request merged**